### DISCUSSION TITLE
Question: Dynamic Changes vanish when load from DB

### DISCUSSION THREAD

**Discussion by codingmachine16** [GrapesJS/grapesjs#3698] [Q&A]
@artf 

Here is my code , I am adding test in a div on select change

        init() {
          this.listenTo(this, "change:noOfLinks", this.getInfoLinks);
        },

        getInfoLinks() {
          let canvasDoc = editor.Canvas.getDocument();
          canvasDoc.getElementById("infoLinks").innerHTML = " test text ";
          
          editor.trigger("change:selectedComponent");
          editor.TraitManager.getTraitsViewer().render();
        },

You can see script is working fine and "test text" is displaying

![image](https://user-images.githubusercontent.com/87995581/129057723-ed78aae9-d2dd-4cee-918d-686f9b66b413.png)

When I press save button. It saves in DB and when I refresh editor, it loads data but the changes "test text" is not there

![image](https://user-images.githubusercontent.com/87995581/129057923-bdef71ce-6e70-4e2d-8a74-3909174a5b20.png)

Please help what I am missing in the code. Thanks

**Comment by artf** [comment]
You have to change the Component Model and not the DOM elements, please read carefully about [Components](https://grapesjs.com/docs/modules/Components.html)

**Reply by codingmachine16** [reply]
Got it . Thanks alot

**Comment by codingmachine16** [comment]
@artf 

I have tried to update model, it is updating every thing expect content 

          const el = wrapper.find("#infoLinks")[0];
          el.set({ tagName: "a" });
          el.set({ type: "link" });
          el.set({ style: { "background-color": "red" } });
          el.set({ content: "test text" });
          el.set({ attributes: { title: "Hello!" } });

This line is not updating 
          el.set({ content: "test text" });
please suggest a solution. Thanks

**Reply by artf** [reply]
do this `el.components('test text')`

**Reply by codingmachine16** [reply]
Thanks alot @artf 

I did 
          el.append("Test text");  and it is working fine.
